mod-source-record-manager-server/src/main/java/org/folio/services/ChangeEngineServiceImpl.java

```diff
@@ -371,6 +372,8 @@ private void populateError(Record record, JobExecution jobExecution, OkapiConnec
       .withContent(record.getRawRecord().getContent())
       .withDescription(new JsonObject().put(MESSAGE_KEY, HOLDINGS_004_TAG_ERROR_MESSAGE).encode())
     );
+    okapiParams.getHeaders().set(RECORD_ID, record.getId());
+
     sendEventToKafka(okapiParams.getTenantId(), Json.encode(eventPayload), DI_ERROR.value(),
       KafkaHeaderUtils.kafkaHeadersFromMultiMap(okapiParams.getHeaders()), kafkaConfig, key)
       .onFailure(th -> LOGGER.error("Error publishing DI_ERROR event for MARC Holdings record with id {}", record.getId(), th));
```

Comment:
Will be better to add this header directly to the KafkaHeaders list instead of okapiParams. I think so because `KafkaHeaderUtils.kafkaHeadersFromMultiMap` method could be changed with for example filtering headers by some params. In this case, your change will not work as expected. 
I suggest extracting variable from `KafkaHeaderUtils.kafkaHeadersFromMultiMap(okapiParams.getHeaders())` and add `recordId` header to this collection of Kafka headers.

Reply:
Done in 1ae9a0b